src/core/tests/type_prop/unique.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <array>
+
+#include "gtest/gtest.h"
+#include "openvino/opsets/opset10.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ov;
+
+namespace {
+constexpr const size_t NUM_OUTPUTS = 4u;
+
+void ASSERT_OUTPUT_SHAPES(const std::shared_ptr<ov::Node>& op, std::array<PartialShape, NUM_OUTPUTS> expected_shapes) {
+    for (size_t i = 0; i < NUM_OUTPUTS; ++i) {
+        EXPECT_TRUE(op->get_output_partial_shape(i).same_scheme(expected_shapes[i]))
+            << "The output shape " << i << " of Unique is incorrect. Expected: " << expected_shapes[i]
+            << ". Got: " << op->get_output_partial_shape(i);
+    }
+}
+
+void ASSERT_ELEMENT_TYPES(const std::shared_ptr<ov::Node>& op, std::array<element::Type, NUM_OUTPUTS> expected_types) {
+    for (size_t i = 0; i < NUM_OUTPUTS; ++i) {
+        EXPECT_EQ(op->get_output_element_type(i), expected_types[i])
+            << "The output element type " << i << " of Unique is incorrect. Expected: " << expected_types[i]
+            << ". Got: " << op->get_output_element_type(i);
+    }
+}
+}  // namespace
+
+TEST(type_prop, unique_no_axis_3d) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape{2, 4, 2});
+    const auto unique = make_shared<opset10::Unique>(data);
+
+    ASSERT_ELEMENT_TYPES(unique, {{element::f32, element::i64, element::i64, element::i64}});
+    ASSERT_OUTPUT_SHAPES(
```

Comment:
Just comment not request.
If there is a helper which give output shapes as container then can use gtest macro

EXPECT_THAT(my_shapes, ElementsAre(...)) or EXPECT_THAT(my_shapes, ElementsAreArray(exp_shapes))

Reply:
Unfortunately we can't retrieve all output shapes this way and I'd rater not create an intermediate  container here. Thanks for the tip though.